File: Algorithm_BinarySearch.cpp
int bsearch(int a[], int left, int right, int x)
{
    while (left <= right)
    {
        int mid = (left + right) >> 1;
        if (a[mid] == x)
            return mid;
        if (a[mid] < x)
            left = mid + 1;
        else
            right = mid - 1;
    }
    return -1;
}

File: Algorithm-BinarySearch.py
# binary search x between a[left..right]
def fun(s,le,ri):
    while le<=ri:
        mid = (le+ri)//2
        if a[mid]==s:
            return mid
        if a[mid]<s:
            le = mid+1
        else:
            ri = mid-1
    return -1
